common/lib/azure/storage/common/core/http_client.rb

```diff
@@ -48,6 +48,10 @@ def reset_agents!
       def build_http(uri)
         ssl_options = {}
         if uri.is_a?(URI) && uri.scheme.downcase == "https"
+          ssl_options[:version] = self.ssl_version if self.ssl_version
+          # min_version and max_version only supported in ruby 2.5
```

Comment:
>only supported in ruby 2.5 [](start = 40, length = 26)

What will happen if it is passed to Ruby version older than 2.5? Just skip it or return a argument error?

Reply:
one NoMethodError would be thrown. if caller specified the min_version.

Reply:
Please see an example [here](https://github.com/cyberark/conjur/blob/master/config/initializers/openssl.rb).

---
In reply to: [223542634](https://github.com/Azure/azure-storage-ruby/pull/130#discussion_r223542634) [](ancestors = 223542634)

Reply:
did not very catch your point, you mean the ruby version should be checked first, and then raise the Error early, or help set the ssl_version and min_version for the caller?

the goal I'm trying to reach is customer have the flexibility to set the values as they wish.
we did not change anything for the caller.

Reply:
I mean, if the version is older than 2.5, we just don't set the `min_version` and `max_version`, as the storage users don't know how the parameters will be used.

---
In reply to: [223546842](https://github.com/Azure/azure-storage-ruby/pull/130#discussion_r223546842) [](ancestors = 223546842)

Reply:
ok, then I think we'd better just leave the information in the api document. (added)
check the ruby version at runtime is not very necessary because the caller will get one error they expected.